### ISSUE TITLE
1 vulnerability requires manual review.

### ISSUE THREAD

**Issue by mcandre** [foreversd/forever#1028]
After installing forever, I get a notice from NPM about forever using dependencies with known vulnerabilities:

```console
$ npm audit
                                                                                
                       === npm audit security report ===                        
                                                                                
┌──────────────────────────────────────────────────────────────────────────────┐
│                                Manual Review                                 │
│            Some vulnerabilities require your attention to resolve            │
│                                                                              │
│         Visit https://go.npm.me/audit-guide for additional guidance          │
└──────────────────────────────────────────────────────────────────────────────┘
┌───────────────┬──────────────────────────────────────────────────────────────┐
│ Low           │ Regular Expression Denial of Service                         │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Package       │ braces                                                       │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Patched in    │ >=2.3.1                                                      │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Dependency of │ forever [dev]                                                │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ Path          │ forever > forever-monitor > chokidar > anymatch > micromatch │
│               │ > braces                                                     │
├───────────────┼──────────────────────────────────────────────────────────────┤
│ More info     │ https://npmjs.com/advisories/786                             │
└───────────────┴──────────────────────────────────────────────────────────────┘
found 1 low severity vulnerability in 2776 scanned packages
  1 vulnerability requires manual review. See the full report for details.
```

**Comment by MetaGauravSaini** [comment]
Help! I'm also facing the same issue.

**Comment by andrewhodel** [comment]
Will this get fixed?  Every project on Github that includes forever and commits a package-lock.json gets a Security Alert about it.

**Comment by kibertoad** [comment]
@indexzero Pretty please?.. It's been fixed on forever-monitor for quite some time already. Could you please either publish it or give me publish permissions?..

**Comment by tilleryd** [comment]
@indexzero It looks like @kibertoad is willing to help out. I'd be happy to as well.

**Comment by kibertoad** [comment]
@tilleryd This was fixed in forever-monitor already, I think: https://github.com/foreversd/forever-monitor/commit/6dfc0e6fbea30616a58374895a36bb27b144e6fb

what is remaining is, unfortunately outside of what you or me can do: forever-monitor needs to be released (I can't do that). Then I could bump dependency version in forever. Then forever would need to be released (I can't do that). So we need @indexzero for that, but he is extremely busy person, so we are where we are :D.

**Comment by andrewhodel** [comment]
It’s like an inverted static library argument!

Sure would be normal if a company needed something like this put one of their paid developers on making something new or forking this if the original dev/team is too busy.

**Comment by kibertoad** [comment]
@andrewhodel Frankly, at this point I would strongly encourage anyone still using forever to switch to pm2 (for more complex loads) or nodemon (for simple cases or dev envs).

**Comment by andrewhodel** [comment]
LMFAO, here is a testimonial about a process manager named pm2. 

I made my Node.js application performance increase by a factor up to 5 without any code change!

Frank Lonas



That has to be a joke, it’s on their website. Batteries forever be damned or what?

**Comment by kibertoad** [comment]
@andrewhodel It's a bit more complicated than that. Forever is a very simple tool that basically does two things: restarts your application and provides basic log management. That's it.
What pm2 offers on top of that is request processing parallelization, which while doesn't increases performance by itself, enables much better horizontal scaling if you have both hardware and software architecture appropriate for that. The idea is this: pm2 can do request load balancing across multiple CPU cores for you, so if you have multiple cores and your software runtime logic does not depend on having single stateful instance (e. g. nothing will break if two threads access same external dependency concurrently), you are going to see significant performance improvements.

**Comment by andrewhodel** [comment]
Yea but that concurrency at the protocol level is only on a single machine, and I am using tcp rather than http for requests for a product so the expectation of normal proxying of requests does not work. 

There are all kinds of reasons for tcp rather than http and session handling with web sockets ends up being a battery hog on this client. 

Plus everyone would forget about it and who would care to help?

You know, like when the Germans stopped creating words because things were better expressed with proofs.

**Comment by kibertoad** [comment]
@andrewhodel You can disable concurrency on pm2 if you want, and you still get a modern, well-supported, production-grade process manager.

**Comment by andrewhodel** [comment]
Or they have a bug on the implementation for this client and it will take 6 months to fix it because they are dead, rather than 1 week and that costs me this much in a frustrated client I want to continue caring about because he actually pays me.

**Comment by kibertoad** [comment]
@andrewhodel Considering that pm2 is backing enterprise offering of their authors, that is an unlikely scenario.

**Comment by andrewhodel** [comment]
Well, IPv6 and anycast video streaming probably aren’t going to go with http proxying for live video and I think it’s important for enterprise to keep up with the underlying protocols that provide them a protocol to work over in the first place. 

I don’t see a reason to have threading in a process launcher, I honestly think it should be a node flag. 

I don’t think Node should have thought of it first, because they were busy on other things.

**Comment by zubairfazal** [comment]
I would recommend using pm2

**Comment by kibertoad** [comment]
Fixed in 2.0.0
